zerver/views/streams.py

```diff
@@ -288,7 +290,18 @@ def add_subscriptions_backend(
         authorization_errors_fatal: bool=REQ(validator=check_bool, default=True),
 ) -> HttpResponse:
     stream_dicts = []
+    color_map = {}
     for stream_dict in streams_raw:
+        # 'color' field is optional
+        # check for its presence in the streams_raw first
+        if 'color' in stream_dict:
+            # check if the provided color has the correct hex code format
+            check_color_format = re.search("^#([0-9a-fA-F]{3,6})$", stream_dict['color'])
+            if (check_color_format):
+                color_map[stream_dict['name']] = stream_dict['color']
```

Comment:
I think this was just solved in #11271 
Should I wait for it to get merged and then update my code accordingly?

Reply:
I just merged that; you can rebase onto master to get it.  Though it would also have been reasonable to just add the commits from #11271 to be the first commits on your branch.